spiderfoot/correlation.py

```diff
@@ -735,71 +894,97 @@ def create_correlation(self, rule: dict, data: list, readonly=False) -> bool:
 
         return True
 
-    # Syntax-check rules
     def check_ruleset_validity(self, rules: list) -> bool:
```

Comment:
The changes from here to end of the file are simply splitting the `check_ruleset_validity` function into two functions:

* check_ruleset_validity
* check_rule_validity

The former loops through rules and validates them with the latter.

This remove some of the levels of nesting and offers a function for validating a single rule, rather than all rules, if desired.